src/executor/infer_graph_attr_pass.cc

```diff
@@ -50,7 +50,15 @@ bool ApplyOpInferAttr<int, FInferStorageType>(const nnvm::Graph& g,
                                               std::vector<int>* out_attrs,
                                               DispatchMode* dispatch_mode) {
   const DevMaskVector& dev_masks = g.GetAttr<DevMaskVector>("dev_mask");
-  return finfer(attrs, dev_masks[nid], dispatch_mode, in_attrs, out_attrs);
+  const bool success = finfer(attrs, dev_masks[nid], dispatch_mode, in_attrs, out_attrs);
+  if (!success) {
```

Comment:
InferShape and InferType would run multiple times if they fail at one time until no more information can be deducted. Does this also apply to InferStorageType or it only infers for one time?

Reply:
Current `finferstorage` returns false only on the cases where some ops are not implemented at all, such as `_sparse_retain(csr,csr)`. InferStorageType only happens once